src/backend/distributed/commands/sequence.c

```diff
@@ -747,7 +772,7 @@ PostprocessGrantOnSequenceStmt(Node *node, const char *queryString)
 		ObjectAddress sequenceAddress = { 0 };
 		Oid sequenceOid = RangeVarGetRelid(sequence, NoLock, false);
 		ObjectAddressSet(sequenceAddress, RelationRelationId, sequenceOid);
-		EnsureDependenciesExistOnAllNodes(&sequenceAddress);
+		EnsureAllObjectDependenciesExistOnAllNodes(list_make1(&sequenceAddress));
```

Comment:
I don't think we should ever pass a reference of a local variable to a list. The lifetime of `sequenceAddress` is scoped to the end of `PostprocessGrantOnSequenceStmt`.

Here, we are not in danger, because the list is used within `PostprocessGrantOnSequenceStmt`. However, as a better practice, please allocate the object addresses via palloc, and set its values, such as in `GetDistributedObjectAddressList`.

Can you go over your patch to see if we have any other such cases? I can see at least `PostprocessAlterStatisticsOwnerStmt `